### PR TITLE
OAK-10933 : skip timestamp calculation if fullgc is not enabled

### PR DESCRIPTION
**Pull request by stefan-egli**

### CODE REVIEW

**Comment by stefan-egli** [comment]
PS: waiting for test to confirm no regression introduced by this - but ideally this would be included asap

**Comment by stefan-egli** [comment]
tests look "good"

**Comment by rishabhdaim** [comment]
IMO, this is an incomplete fix. We need to add `fullGCEnabled` check before `isFullGCDryRun` at https://github.com/apache/jackrabbit-oak/blob/trunk/oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGCRecommendations.java#L152 & https://github.com/apache/jackrabbit-oak/blob/trunk/oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGCRecommendations.java#L152, just like we did here: https://github.com/apache/jackrabbit-oak/blob/trunk/oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGCRecommendations.java#L304

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGCRecommendations.java [hunk 1]
```diff
@@ -170,7 +170,7 @@ public class VersionGCRecommendations {
             } else {
                 oldestModifiedDryRunDocTimeStamp.set(fullGCDryRunTimestamp);
             }
-        } else {
+        } else if (fullGCEnabled) {
```
Comment (rishabhdaim) [inline oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGCRecommendations.java:173]: We should move this check before the `isFullGCDryRun` check.

**Review by rishabhdaim — CHANGES_REQUESTED** [review]

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGCRecommendations.java (hunk 1 above)
Comment (stefan-egli) [reply oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGCRecommendations.java:173]: I was assuming dry run mode wouldn't be enabled if fullgc is disabled - but probably better to protect from that indeed. now done in https://github.com/apache/jackrabbit-oak/pull/1570/commits/0777573384e8d6504ab06eda073d0805a4e8a53a

**Comment by stefan-egli** [comment]
agree this is incomplete - it is merely a first fix for the most noisy logger. we likely will have to exclude more - but that shouldn't prevent this one from going through.

**Review by rishabhdaim — APPROVED** [review]

**Pull request merged**